src/sentry/static/sentry/app/components/events/interfaces/spans/span_tree.tsx

```diff
@@ -0,0 +1,665 @@
+
+            this.toggleSpanTree();
+          }}
+        >
+          <span style={{marginRight: '2px', textAlign: 'center'}}>
+            <Count value={numOfSpanChildren} />
+          </span>
+          <div style={{marginRight: '2px', width: '5px', textAlign: 'right'}}>
+            {chevron}
+          </div>
+        </SpanTreeToggler>
+      </React.Fragment>
+    );
+  };
+
+  renderTitle = () => {
+    const {span, treeDepth} = this.props;
+
+    const op = span.op ? <strong>{`${span.op} \u2014 `}</strong> : '';
+    const description = _.get(span, 'description', span.span_id);
+
+    const MARGIN_LEFT = 8;
+    const left = treeDepth * (8 * 0) + MARGIN_LEFT;
+
+    return (
+      <SpanBarTitleContainer>
+        {this.renderSpanTreeToggler({left})}
+        <SpanBarTitle
+          data-component="span-bar-title"
+          style={{
+            left: `${left}px`,
+            width: '100%',
+          }}
+        >
+          <span>
+            {op}
+            {description}
+          </span>
+        </SpanBarTitle>
+      </SpanBarTitleContainer>
+    );
+  };
+
+  renderSpanChildren = () => {
+    if (!this.state.showSpanTree) {
+      return null;
+    }
+
+    return this.props.renderedSpanChildren;
+  };
+
+  render() {
+    const {span, spanBarColour} = this.props;
+
+    const start_timestamp: number = span.start_timestamp;
+    const end_timestamp: number = span.timestamp;
+
+    const duration = (end_timestamp - start_timestamp) * 1000;
+    const durationString = `${duration.toFixed(3)} ms`;
+
+    const bounds = this.getBounds();
+
+    const isVisible = bounds.end > 0 && bounds.start < 1;
+
+    return (
+      <React.Fragment>
+        <SpanRow
+          data-span-hidden={isVisible ? 'false' : 'true'}
+          style={{
+            display: isVisible ? 'block' : 'none',
+            boxShadow: this.state.displayDetail ? '0 -1px 0 #d1cad8' : void 0,
+          }}
+          onClick={() => {
+            this.toggleDisplayDetail();
+          }}
+        >
+          <SpanBar
+            data-span="true"
+            style={{
+              backgroundColor: spanBarColour,
+              left: toPercent(bounds.start),
+              width: toPercent(bounds.end - bounds.start),
+            }}
+          />
+          {this.renderTitle()}
+          <Duration>{durationString}</Duration>
+          {this.renderDetail({isVisible})}
+        </SpanRow>
+        {this.renderSpanChildren()}
+      </React.Fragment>
+    );
+  }
+}
+
+const TraceViewContainer = styled('div')`
+  overflow-x: hidden;
+  border-bottom-left-radius: 3px;
+  border-bottom-right-radius: 3px;
+`;
+
+const SPAN_ROW_HEIGHT = 25;
+
+const SpanRow = styled('div')`
+  position: relative;
+  overflow: hidden;
+
+  cursor: pointer;
+  transition: background-color 0.15s ease-in-out;
+
+  &:last-child {
+    & > [data-component='span-detail'] {
+      border-bottom: none !important;
+    }
+  }
+
+  & > [data-span='true'] {
```

Comment:
Were you not able to apply these styles with props to styled components?

Reply:
done